### PR TITLE
Upgrade pdfboxing to 0.1.14

### PR DESCRIPTION
**Pull request by mrichards42**
Including an upgrade to PDFBox from 1.8.10 to 2.0.16
PDFBox 2 includes a new function: PDAcroForm.flatten, which can replace
our entire (probably a bit flaky) form-filling code.

### CODE REVIEW

File: src/form_works/render.clj [hunk 1]
```diff
@@ -1,94 +1,25 @@
...
-      (.clear (.getFields form))
-      (-> doc
-          .getDocumentCatalog
-          .getAcroForm
-          .getDictionary
-          (.getDictionaryObject "Fields")
-          .clear)
-      (.save doc output))))
+        (fill-field form (key field) (val field)))
+      (.refreshAppearances form)
```
Comment (mrichards42) [inline src/form_works/render.clj]: This call to `refreshAppearances` (also new in PDFBox 2) should allow us to use auto-sized text fields. Though NB: PDFBox only supports auto-sizing in single line text fields, not multiline fields.

**Comment by dzcode** [comment]
Does this need to be merged ahead of NVRD? Just my 2c, but in the interest of reducing new changes in case something isn't working right, would it make sense to hold off until after?

File: src/form_works/render.clj [hunk 2]
```diff
@@ -1,94 +1,25 @@
...
+      (.flatten form)
+      (.save doc output)
+      (.close doc))))
```
Comment (gws) [inline src/form_works/render.clj]: I think because of the `with-open` that this'll get closed twice. May just be useless if `(.close doc)` is idempotent.

**Comment by mrichards42** [comment]
@dzcode good questions -- we have some partners w/ campus addresses that are _way_ too long for this form, so they end up getting cut off, at best producing an invalid form. Since we're about to have a lot of signups (and partners will be doing reg drives), product wants to try to get this out asap.

File: src/form_works/render.clj (hunk 2 above)
Comment (mrichards42) [reply src/form_works/render.clj]: Yup, you're totally right.

**Pull request merged**